python/_dynet.pyx

```diff
@@ -97,20 +97,21 @@ cdef class DynetParams: # {{{
             shared_parameters([type]): [description] (default: None)
         """
         cpu_use = False
-        if '--dynet-gpu' in sys.argv:
-            sys.argv.remove('--dynet-gpu')
-            sys.argv.append('--dynet-gpus')
-            sys.argv.append('1')
-        elif not ('--dynet-gpus' in sys.argv or
-                  '--dynet-devices' in sys.argv or
-                  '--dynet-viz' in sys.argv):
+        sys_argv = sys.argv
```

Comment:
Just out of curiosity, did you test this? I thought you would have to do `sys_argv = list(sys.argv)` to make a copy of the list, but I might be wrong.

Reply:
ohm you are right!